blaze/expr/arithmetic.py

```diff
@@ -50,18 +50,38 @@ def name(o):
         return None
 
 
-class BinOp(ElemWise):
-    __slots__ = '_hash', 'lhs', 'rhs'
+class Op(ElemWise):
+    __slots__ = '_hash', 'operands'
+    __inputs__ = 'operands',
+
+    def __init__(self, operands):
+        self._operands=operands
+
+    @property
+    def _inputs(self):
+        result = []
+        for _op in self._operands:
+            if isinstance(_op, Expr):
+               result.append(_op)
+        return tuple(result)
```

Comment:
I think you can bypass the list -> tuple conversion and just construct the tuple directly:

```
return tuple(op for op in self._operands if isinstance(op, Expr))
```